spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/MemoryData.scala

```diff
@@ -99,6 +100,17 @@ sealed trait MemoryData extends Serializable {
       case _ => throw new UnsupportedOperationException(s"unsupported data type")
     }
   }
+
+  private def checkConsistency(shape: Array[Int], layout: Int): Unit = {
+    // TODO current only handle the 4-D and 2-D shape
```

Comment:
What will be `TODO`

Reply:
I want to check the shape is consistent with layout. Currently, it only implements the invalidation check.